FirebaseSessions/Sources/FirebaseSessions.swift

```diff
@@ -26,13 +28,34 @@ protocol SessionsProvider {
 @objc(FIRSessions) class Sessions: NSObject, Library, SessionsProvider {
   // MARK: - Private Variables
 
-  /// The app associated with all sessions.
+  /// The Firebase App associated with Sessions.
   private let googleAppID: String
 
+  /// The Firebase SDKs that Sessions depends on
+  private let installations: Installations
+
+  /// Top-level Classes in the Sessions SDK
+  private let main: Main
+  private let initiator: Initiator
+  private let identifiers: Identifiers
+
   // MARK: - Initializers
 
   required init(app: FirebaseApp) {
     googleAppID = app.options.googleAppID
+
+    installations = Installations.installations(app: app)
+
+    identifiers = Identifiers(installationIDProvider: installations)
+    main = Main(identifiers: identifiers)
+    initiator = Initiator()
+
+    super.init()
+
+    initiator.beginListening {
+      self.identifiers.generateNewSessionID()
+      self.main.runMain()
+    }
```

Comment:
1. Who will be calling `required init(app: FirebaseApp)`? SDKs that use the Sessions SDK?

---

2.  I think I would treat the `init(app:)` as a convenience initializer and call into an internal initializer to inject the class members. This will help will testability. 

Something like:

```swift
  required convenience init(app: FirebaseApp) {
    // ... 
  }
    
  init(appID: String, installations: Installations, initiator: Initiator, /* ... */) {  
    // ... 
  }
    
```

--- 

3. Additionally, unrelated to this PR, but why is `init(app:)` marked as `required`? Are there plans to subclass the `Sessions` class?

Reply:
Here's where it's called, and where I got the idea to add required:

![Screen Shot 2022-10-04 at 11 04 54 AM](https://user-images.githubusercontent.com/555046/193855480-4414d118-26bb-4a40-9947-0337bb01417d.png)

Reply:
Updated to match the recommendations